### PR TITLE
Multiregional Phone Number Recognizer

### PR DESCRIPTION
**Pull request by SharonHart**
## Change Description

Add multi regional phone number recognizer.
Uses python-phonenumbers package to match text against multiple regions' regex patterns, as well as finding international patterned phone number.

I've used the context enhancement with 'phone number' supportive words, along with the region code and description to raise confidence of one region compared to others.

We need to decide which phone numbers we want to support be default. Setting all supported regions returned 125 matches for a random phone number for example, which adds significant noise to the returned entities.

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by omri374** [comment]
@SharonHart awesome work! Thanks for this. Left some comments mainly around customizability capabilities

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    CONTEXT = ["phone", "number", "telephone", "cell", "cellphone", "mobile", "call"]
+    DEFAULT_SUPPORTED_COUNTRY_CODES = ("US", "UK", "DE", "FE", "IL")
+
+    def __init__(
+        self,
+        supported_language: str = "en",
+        supported_entities: List[str] = [
+            code + ENTITY_TYPE_SUFFIX for code in DEFAULT_SUPPORTED_COUNTRY_CODES
+        ]
+        + [INTERNATIONAL_ENTITY_TYPE],
```
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py:33]: That way, there wouldn't be any option to opt it out. I can add a flag for adding it, works?

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ class UsPhoneRecognizer(PatternRecognizer):
         Pattern("Phone (weak)", r"(\b\d{10}\b)", 0.05),
```
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py:23]: Yes, added another context word

File: presidio-analyzer/Pipfile [hunk 3]
```diff
@@ -10,6 +10,7 @@ tldextract = "*"
 numpy = "==1.19.3"
 flask = "==1.1.2"
 pyyaml = "*"
+phonenumbers = "*"
```
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-analyzer/Pipfile:13]: yes just did it in a local commit

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        Iterates over entities, fetching regions, then matching regional
+        phone numbers patterns against the text.
+        :param text: Text to be analyzed
+        :param entities: Entities this recognizer can detect
+        :param nlp_artifacts: Additional metadata from the NLP engine
+        :return: List of phone numbers RecognizerResults
+        """
+        results = []
+        for entity in entities:
+            region = entity.replace(ENTITY_TYPE_SUFFIX, "")
```
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py:69]: Checked, you're right. Added a check against the package SUPPORTED_REGIONS

**Comment by SharonHart** [comment]
@omri374 Resolved your comments, left one question

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (omri374) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py:23]: But do we need an external us phone number recognizer if we have the new one which covers probably more patterns?

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (omri374) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py:33]: Yep

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/us_phone_recognizer.py:23]: We don't. I plan to add this recogznier as default in a separate PR to reduce file changes, will remove the US one in the same PR.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py]: Done

**Review by omri374 — APPROVED** [review]

**Review by navalev — APPROVED** [review]

**Comment by Oleg-Litvinov-EPAM** [comment]
Hi everyone!

Why this recognizer is not added to the list of the ones loaded by default?

**Comment by omri374** [comment]
Hi @Oleg-Litvinov-EPAM, good question! tagging @SharonHart as we had a discussion on this before.

**Comment by SharonHart** [comment]
@Oleg-Litvinov-EPAM This is the plan eventually. Next phase before doing so is to unify all regional entities to a single PHONE_NUMBER entity and aggregate all possible matches by start and end indicies.

**Comment by Oleg-Litvinov-EPAM** [comment]
@SharonHart, this really makes sense. Thank you very much for the clarification!)

**Pull request merged**